core/io/marshalls.cpp

```diff
@@ -1068,6 +1068,9 @@ Error encode_variant(const Variant &p_variant, uint8_t *r_buffer, int &r_len, bo
 			}
 		} break;
 		default: {
+#ifdef REAL_T_IS_DOUBLE
+			flags |= ENCODE_FLAG_64;
+#endif
```

Comment:
```suggestion
		} break;
#ifdef REAL_T_IS_DOUBLE
		case Variant::VECTOR2:
		case Variant::RECT2:
		case Variant::VECTOR3:
		case Variant::TRANSFORM2D:
		case Variant::PLANE:
		case Variant::QUATERNION:
		case Variant::AABB:
		case Variant::BASIS:
		case Variant::TRANSFORM3D:
		case Variant::PACKED_VECTOR2_ARRAY:
		case Variant::PACKED_VECTOR3_ARRAY: {
			flags |= ENCODE_FLAG_64;
		} break;
#endif // REAL_T_IS_DOUBLE
		default: {
```

Reply:
I dont think `PACKED_FLOAT64_ARRAY` and `PACKED_INT64_ARRAY` need that flag